tests/e2e/utils/product-editor.js

```diff
@@ -10,6 +10,243 @@ import * as api from './api';
 
 const REGEX_URL_PRODUCTS = /\/wc\/v3\/products\/\d+(\/variations\/\d+)?\?/;
 
+async function setAttributeValue( locator, value ) {
+	const tagName = await locator.evaluate( ( element ) => element.tagName );
+
+	if ( tagName === 'SELECT' ) {
+		await locator.selectOption( value );
+	} else {
+		await locator.fill( value );
+	}
+
+	await expect( locator ).toHaveValue( value );
+}
+
+/**
+ * Gets E2E test utils for facilitating writing tests for the classic product editor.
+ *
+ * @param {Page} page Playwright page object.
+ */
+export function getClassicProductEditorUtils( page ) {
+	const locators = {
+		getPluginTab() {
+			return page.locator( '.gla_attributes_tab' );
+		},
+
+		getPluginPanel() {
+			return page.locator( '#gla_attributes' );
+		},
+
+		getPluginVariationMetaBox() {
+			return page.locator( '.gla-metabox:visible' ).first();
+		},
+
+		getChannelVisibilityMetaBox() {
+			return page.locator( '#channel_visibility' );
+		},
+
+		getChannelVisibilityHeading() {
+			return this.getChannelVisibilityMetaBox().getByRole( 'heading', {
+				name: 'Channel visibility',
+			} );
+		},
+
+		getProductAttributesHeading() {
+			return this.getPluginPanel().getByRole( 'heading', {
+				name: 'Product attributes',
+			} );
+		},
+
+		getChannelVisibility() {
+			const metaBox = this.getChannelVisibilityMetaBox();
+
+			return {
+				selection: metaBox.getByRole( 'combobox' ),
+				help: metaBox.locator( '.description' ),
+				notice: metaBox.locator( '.sync-status' ),
+				status: metaBox.locator( '.sync-status p' ).nth( 1 ),
+				issues: metaBox.getByRole( 'listitem' ),
+			};
+		},
+
+		getSelectWithTextInput() {
+			const field = page.locator( '.select-with-text-input' ).first();
+
+			return {
+				selection: field.getByRole( 'combobox' ),
+				input: field.getByRole( 'textbox' ),
+			};
+		},
+
+		getDateAndTimeInputs( locator = page ) {
+			const simple = locator.locator(
+				'.gla_attributes_availabilityDate_field'
+			);
+			const variation = locator.locator(
+				'.gla_variation_attributes\\[0\\]_availabilityDate_field'
+			);
+			const field = simple.or( variation );
+
+			return {
+				dateInput: field.locator( 'input[type=date]' ),
+				timeInput: field.locator( 'input[type=time]' ),
+			};
+		},
+
+		getMultipackInput() {
+			return page.locator( '.gla_attributes_multipack_field input' );
+		},
+
+		getAllProductAttributes( locator = page ) {
```

Comment:
I see this function is 99% the same as the `getProductBlockEditorUtils` version. Can we refactor it?

Reply:
Adjusted in https://github.com/woocommerce/google-listings-and-ads/pull/2363/commits/8a7993308650aae932702368c37b2e9160da2922.